OSDOCS-20655: update intro with istio details (#115347)
openshift/openshift-docs: modules/con-about-connectivity-link.adoc

Apply policies to standard link:https://gateway-api.sigs.k8s.io/[Gateway API] resources in {ocp} clusters by using {prodname} and remove the need to embed networking code into your applications. This enables an infrastructure-as-code approach to ingress traffic management and protocol support.

> You can use {service-mesh} {service-mesh-version} or configure the OpenShift `istio` controller as the Gateway API provider.

Reviewer: I would add a line describing how to install OSSM3 yourself. Probably link to https://docs.redhat.com/en/documentation/red_hat_openshift_service_mesh/3.3/html-single/installing/index

> You can use an `istio` controller by enabling the Gateway API with a `GatewayClass` object that specifies the `openshift-default` namespace. See "Enable Gateway API for the Ingress Operator" in the Additional resources section.

Reviewer: ```suggestion
You can use an `istio` controller by enabling the Gateway API by creating a specific `GatewayClass` object that specifies the `openshift.io/gateway-controller/v1` controller name. See "Enable Gateway API for the Ingress Operator" in the Additional resources section for the enabling procedure.
```

[IMPORTANT]
====

> If you want to use {service-mesh} and mTLS, but keep control over your Istio configuration, you must configure and apply deploy your own `Istio` CR and `IstioCNI` CRs.

Reviewer: ```suggestion
If you want to use {prodname} with mTLS enabled or keep control over Istio configuration, you must deploy {service-mesh} {service-mesh-version} and apply your own `Istio` CR and `IstioCNI` CRs.
```

> If are not using {service-mesh} and you require mutual TLS (mTLS), you must create a custom `Istio` custom resource (CR) with the Container Network Interface (CNI) requirement enabled. You can also enable Gateway API on your custom `Istio` CR. When defining your `Gateway` objects, ensure that they avoid using the `openshift.io/gateway-controller/v1` controller name. This prevents the Cluster Ingress Operator (CIO) from attempting to manage resources for your custom `Istio` control plane.

Reviewer: I checked for ocp421 and ocp422. Because of https://redhat.atlassian.net/browse/NE-2286 the way the controller works in background changed so its no longer possible to just create Istio CR with own configuration.

I would suggest to user to either install OSSM3 and deploy their own Istio CR and IstioCNI CR if they want to use mTLS and have control over Istio configuration, or configure the `istio` controller the way Openshift docs say. It is possible to combine the approaches but for clarity I would present this as "either-or" choice, than there is no possibility for conflicts.
